frontend/openchat-client/src/openchat.ts

```diff
@@ -911,12 +911,98 @@ export class OpenChat extends EventTarget {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canDeleteOtherUsersMessages(chat);
+    }
+
+    canPinMessages(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canPinMessages(chat);
+    }
+
+    canReactToMessages(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canReactToMessages(chat);
+    }
+
+    canReplyInThread(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canReplyInThread(chat);
+    }
+
+    canSendMessages(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canSendMessages(chat, this._liveState.userStore);
+    }
+
+    canChangeRoles(chatId: string, currentRole: MemberRole, newRole: MemberRole): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat?.kind === "group_chat" && canChangeRoles(chat, currentRole, newRole);
+    }
+
+    canUnblockUsers(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canUnblockUsers(chat);
+    }
+
+    canRemoveMembers(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canRemoveMembers(chat);
+    }
+
+    canEditGroupDetails(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canEditGroupDetails(chat);
+    }
+
+    canChangePermissions(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat?.kind === "group_chat" && canChangePermissions(chat);
+    }
+
+    canInviteUsers(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canInviteUsers(chat);
+    }
+
+    canDeleteGroup(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canDeleteGroup(chat);
+    }
+
+    canMakeGroupPrivate(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canMakeGroupPrivate(chat);
+    }
+
+    canLeaveGroup(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canLeaveGroup(chat);
+    }
+
+    canAddMembers(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && canAddMembers(chat);
+    }
+
+    isPreviewing(chatId: string): boolean {
+        const chat = this._liveState.chatSummaries[chatId];
+        return chat !== undefined && isPreviewing(chat);
+    }
```

Comment:
Could you DRY this up a bit with something like: 

```
    chatPermits(chatId: string, predicate: (chat: ChatSummary) => boolean): boolean {
        const chat = this._liveState.chatSummaries[chatId]; 
        return chat ? predicate(chat) : false;
    }
```

Reply:
Done